velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -321,15 +336,20 @@ bool registerAverageAggregate(const std::string& name) {
         if (exec::isRawInput(step)) {
           switch (inputType->kind()) {
             case TypeKind::SMALLINT:
-              return std::make_unique<AverageAggregate<int16_t>>(resultType);
+              return std::make_unique<
```

Comment:
Shouldn't we update signatures and this logic to add support for decimals?

Reply:
We split templating AverageAggregate and support decimals.
Here is the PR: https://github.com/facebookincubator/velox/pull/2639 that specializes for decimals.